### PR TITLE
Attempt to use firebase 3.0

### PR DESCRIPTION
**Pull request by rogeriochaves**
So, first of all, I really needed this project going, so I decided to help, then I noted that this is *a lot* to maintain, congratulations for the awesome work!

What I did was changing the example and use it to test if the stuff worked. Auth was not working, so I needed to upgrade to Firebase 3.0.

But I've had some problems upgrading the javascript, because firebase 3.0 doesn't work when bundled inside something that uses `"use strict"`, as elm does, check it out: https://groups.google.com/forum/#!topic/firebase-talk/gFtK1YoRUoE

So I've decided to delete firebase and rely on the global available one, and found out that this might be a much better approach, because:
- Firebase is **HUGE**, it weights 300kb minified, so it is indeed better to serve it separately, from a CDN, or at least to have a separate chunk for it if you are using webpack or something, so it can load in parallel
- I can upgrade firebase for non-breaking changes without changing elmfire

Going forward, I've realized that the API must change, we have to pass more data than the database url (apiKey, authDomain, databaseURL, storageBucket, messagingSenderId now), and we have to instantiate it without child paths, we could have something like:

```elm
firebase : FirebaseInstance
firebase =
    initializeApp
        { apiKey = "foo"
        , authDomain = "bar.firebaseapp.com"
        , databaseURL = "https:/bar.firebaseio.com"
        , storageBucket = "bar.appspot.com"
        , messagingSenderId = "1000000000"
        }
```

Then, I don't see a reason why the elm api shouldn't just mimic the js api, we could use it like this:
```elm
update : Msg -> Model -> ( Model, Cmd Msg )
update msg model =
    case msg of
        Send text ->
            let
                task =
                    firebase.database.ref "mychildpath"
                        |> .set (JE.string text)
            in
                ( model
                , Task.perform
                    (Sent << Err)
                    (Sent << Ok << (always ()))
                    task
                )
        Login ->
            let
                task =
                    firebase.auth.signInWithEmailAndPassword "test@test.com" "123test"
            in
                ( model
                , Task.perform
                    (Sent << Err)
                    (Sent << Ok << (always ()))
                    task
                )
```

### CODE REVIEW

File: src/Native/ElmFire.js [hunk 1]
```diff
@@ -82,7 +82,13 @@ var _ThomasWeiser$elmfire$Native_ElmFire = function () {
...
       switch (head._0) {
         case 'url':
-          ref = new Firebase (head._1);
+          if (!firebase.apps.length) {
+            firebase.initializeApp({databaseURL: head._1});
+          }
+          ref = firebase.database();
+          break;
+        case 'ref':
+          ref = getRefStep (rest) .ref (head._1);
```
Comment (jamesbvaughan) [inline src/Native/ElmFire.js:91]: Why the spaces after function names?

Comment (rogeriochaves) [reply src/Native/ElmFire.js:91]: idk, that's the way thomas was doing ¯\\\_(ツ)_/¯

**Comment by ThomasWeiser** [comment]
@rogeriochaves  thank you very much for your contribution. Will review it ASAP.

**Comment by rogeriochaves** [comment]
@ThomasWeiser it's not really a valuable code contribution, I opened this PR more for the discussion of what path will we follow

File: src/Native/ElmFire.js (hunk 1 above)
Comment (ThomasWeiser) [reply src/Native/ElmFire.js:91]: That's my style for JS code.
I would appreciate if contributors would stick the style of the existing code.

**Comment by ThomasWeiser** [comment]
As per your suggestion, firebase.js script is now to be included separately. For pros and cons see the [commit msg](https://github.com/ThomasWeiser/elmfire/commit/899551a45d62e563a0c973ebfc5521558f81811a).

Please direct API discussions to #16 or separate issues/PRs.

**Pull request closed without merging**